lib/fizzy/instantiate.cpp

```diff
@@ -199,11 +200,12 @@ std::tuple<bytes_ptr, Limits> allocate_memory(const std::vector<Memory>& module_
             (memory_max.has_value() && *memory_max > memory_pages_limit))
         {
             throw instantiate_error{"cannot exceed hard memory limit of " +
-                                    std::to_string(memory_pages_limit * PageSize) + " bytes"};
+                                    std::to_string(uint64_t{memory_pages_limit} * PageSize) +
```

Comment:
This actually will never overflow, because module memory with min or max > 65536 will be an invalid (malformed) module (so will fail earlier in parser), but I think it's better to cast here, too, for consistency. But no test possible.